docs/contributing/language_guide.rst

```diff
@@ -0,0 +1,57 @@
+software management. This document favors clarity and accuracy over ease of reading.
+
+Notation
+--------
+
+Each paragraph is preceded by a list of terms that are covered for the first time. In the paragraph
+body, terms that correspond to Python code are capitalized based on their type (see PEP 8) and are
+indicated by double backticks \`\`object\`\`. (``ClassName``, ``function_name``).  Terms that are
+not code objects in in bold. (**plugin**)
+
+Definitions
+-----------
+
+[``ContentUnit``, **plugin**, ``type``, ``Artifact``]
+    ``pulpcore`` is a generalized backend with a REST API and a plugin API. Users will also need at
+    least one **plugin** to manage content.  Each **plugin** defines at least one ``type`` of
+    ``ContentUnit`` (like .rpm or .deb), which is the smallest set of data that can be managed by
+    Pulp. The plural form of ``ContentUnit`` is ``ContentUnits``, rather than Content or Units.
+    Files that belong to a ``ContentUnit`` are called ``Artifacts``, and each ``ContentUnit`` can
+    have 0 or many ``Artifacts``.  ``Artifacts`` can be shared by multiple ``ContentUnits``.
+
+[``Repository``, **add**, **remove**, **RepositoryVersion**]
+    ``ContentUnits`` in Pulp are organized by their membership in a ``Repository`` over time. Users
+    can **add** or **remove** ``ContentUnits`` to a ``Repository`` by creating a new
+    ``RepositoryVersion`` and specifying the ``ContentUnits`` to **add** and **remove**.
+
+[**upload**]
+    ``ContentUnits`` can be created in Pulp manually. Users specify the ``Artifacts`` that belong
+    to the ``ContentUnit`` and the **plugin** that defines the ``ContentUnit`` ``type``.
+    ``Artifacts`` that are not already known by Pulp should be **uploaded** to Pulp prior to
+    creating a new ``ContentUnit``. ``ContentUnits`` are manually **added** to a
+    ``Repository`` manually by creating a new ``RepositoryVersion``.
+
+[**external source**, **sync**]
+    Users can fetch ``ContentUnits`` and **add** them to their ``Repository`` by **syncing** with an
+    **external source**. The logic and configuration that specifies how Pulp should to interact
+    with an **external source** is provided by an ``Importer`` and is defined by the same
+    **plugin** that defines that ``type`` of ``ContentUnit`` that the **external source** contains.
+    ``pulpcore`` supports multiple ``sync_modes``, including ``additive`` (``ContentUnits`` are
+    only **added**) and ``mirror`` (``ContentUnits`` are **added** and **removed** to match the
+    **external source**.)
+
+[``hosted``, **metdata**, ``Publisher``, ``Publication``, ``Distribution``]
+    All ``ContentUnits`` that are managed by Pulp are **hosted** as part of the ``pulpcore``
+    Content App. **plugin** defined ``Publishers`` generate ``Publications``, which
+    refer to the **metadata** and ``Artifacts`` of the ``ContentUnits`` in a ``RepositoryVersion``
```

Comment:
How is `**metadata**` different from `PublishedMetadata` as a formal software object? I think introducing both `PublishedMetadata` and `PublishedArtifacts` may be good. What do you think?

Reply:
I wanted to avoid defining any objects that aren't user facing, since they wont be semantically versioned.

Reply:
What about plugin writers as a form of user? Those two objects are in the plugin API so they would be semver governed by the plugin API version. If it's just for users that is OK too, but I figured it would be in the user guide, not the contributor guide.

Reply:
I'm thinking of this as a contributor guide for how to write about user facing concepts. I am +1 on creating a second document in this style in the plugin writers guide

Reply:
Sounds good to me. I'm +1 to merge with the other small grammar fixes.